src/lightning/fabric/fabric.py

```diff
@@ -610,12 +610,17 @@ def init_tensor(self) -> Generator:
             yield
 
     @contextmanager
-    def init_module(self) -> Generator:
+    def init_module(self, empty_weights: Optional[bool] = None) -> Generator:
         """Instantiate the model and its parameters under this context manager to reduce peak memory usage.
 
         The parameters get created on the device and with the right data type right away without wasting memory being
         allocated unnecessarily. The automatic device placement under this context manager is only supported with
         PyTorch 2.0 and newer.
+
+        Args:
+            empty_weights: Whether to initialize the model with empty weights (uninitialized memory).
```

Comment:
Not liking the proposed names personally. Right now, this initial version is "just faster" because it essentially skips the `torch.nn.init` which would normally overwrite all memory with data. With the current empty init, we simply rely on the default `torch.empty` tensor creation in all `torch.nn` layers. This does allocate memory, just doesn't initialize it (the underlying data is garbage). I believe "empty weights" is a fine name, and allows us to change the implementation further in the future (e.g. fake tensor or other ways) without needing to choose a new name. 
Note, accelerate also uses a [similar name](https://huggingface.co/docs/accelerate/v0.19.0/en/package_reference/big_modeling#accelerate.init_empty_weights).